### PR TITLE
Add focus-in event handling on the sb.input

### PR DESCRIPTION
**Pull request by lessless**
This PR adds a `focus-in` handling for the `sb.input`.
Possible use-case is a data preloading before user started typing.

### CODE REVIEW

**Comment by amk221** [comment]
Thank you very much for this, but could you use `{{select-box on-focus-in=preloadThings}}` instead?
That should work, although I need to [document it](https://github.com/amk221/ember-select-box/commit/c0df42dd9c2ecbb402ea93c9dede64a4b6e97189).

_edit_: You can also use `{{sb.input focus=doSomething}}` but due to Ember's attrs proxy that will wipe out any existing `focus()` functionality

**Comment by lessless** [comment]
Hm,

there is a weird thing with `on-focus-in` - it throws an exception inside [jQuery focus-in handler](https://github.com/jquery/jquery/blob/c9935b6d2db9e1be4bed12f7419e98cdca45763e/src/event/focusin.js#L22) when receives same argument as`on-search` does: `on-focus-in = attrs.on-search`,
but working fine with component action `on-focus-in=(action 'preloadThings')`

Anyways, problems seem to be solved, but  I will really appreciate if you can give more context about the suggestion in the edit.

**update:**
there is a subtle nuance - if I just perform search and set `availableThings` to the array of returned entities there is  a need to call `sb.open();` in the `on-focus-in` handler for  wrapper div to obtain `is-open` class.
Otherways it will have only `is-focused` despite the fact that it actually displays the results.

**Comment by amk221** [comment]
For anybody interested the behaviour that @lessless required can be found [here](https://github.com/amk221/-ember-filter-select-example/blob/master/app/components/filter-select.js).

TLDR; You must preventDefault on mouseDown if you don't want focus to be lost
